cohortextractor/backends/base.py

```diff
@@ -10,17 +16,13 @@ def register_backend(backend_class):
 
 
 class BaseBackend:
-    backend_id = NotImplemented
-    query_engine_class = NotImplemented
-    patient_join_column = NotImplemented
+    backend_id: str
+    query_engine_class: type["BaseSQLQueryEngine"]
+    patient_join_column: str
 
     tables = None
 
     def __init_subclass__(cls, **kwargs):
-        assert cls.backend_id != NotImplemented
-        assert cls.query_engine_class != NotImplemented
-        assert cls.patient_join_column != NotImplemented
```

Comment:
I think we've lost something useful here. Can we assert that they're not `None` as a check instead?